Sprint-3/2-practice-tdd/count.test.js

```
@@ -22,3 +21,101 @@ test("should count multiple occurrences of a character", () => {
 // And a character char that does not exist within the case-sensitive str,
 // When the function is called with these inputs,
 // Then it should return 0, indicating that no occurrences of the char were found in the case-sensitive str.
+test("should return 0 if the character does not exist in the string", () => {
+  expect(countChar("hello", "z")).toEqual(0);
+});
+
+// Scenario: Single Occurrence
+// Given the input string str,
+// And a character char that occurs exactly once in str,
+// When the function is called with these inputs,
+// Then it should return 1, indicating one occurrence of char in str.
+test("should count a single occurrence of a character", () => {
+  expect(countChar("hello", "h")).toEqual(1);
+});
+
+// Scenario: Empty String
+// Given an empty input string str,
+// And any character char,
+// When the function is called with these inputs,
+// Then it should return 0, as there are no characters to count.
+test("should return error if the string is empty", () => {
+  expect(countChar("", "a")).toEqual("Error: The string cannot be empty.");
+});
```

This test description and the behavior of the function do not quite match the description in the comment.

---

Where do you get the idea the function should return an error when the first parameter is an empty string?

Shouldn't the comments in the original file serve as the spec?

---

Should be 0. I got myself all confused. I apologize for all these... I should know enough to prevent you having to point out this many issues. I would do better going forward